pkg/kp/rcstore/fake_store.go

```diff
@@ -0,0 +1,134 @@
+package rcstore
+
+import (
+	"strconv"
+
+	"github.com/square/p2/pkg/labels"
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/rc/fields"
+	"github.com/square/p2/pkg/util"
+)
+
+type fakeStore struct {
+	rcs     map[string]*fakeEntry
+	creates int
+}
+
+type fakeEntry struct {
+	fields.RC
+	watchers []chan struct{}
+}
+
+var _ Store = &fakeStore{}
+
+func NewFake() *fakeStore {
+	return &fakeStore{
+		rcs:     make(map[string]*fakeEntry),
+		creates: 0,
+	}
+}
+
+func (s *fakeStore) Create(manifest pods.Manifest, nodeSelector labels.Selector, podLabels labels.Set) (fields.RC, error) {
+	// A real replication controller will use a UUID.
+	// We'll just use a monotonically increasing counter for expedience.
+	s.creates += 1
+	id := strconv.Itoa(s.creates)
+
+	entry := fakeEntry{
+		RC: fields.RC{
+			Id:              id,
+			Manifest:        manifest,
+			NodeSelector:    nodeSelector,
+			PodLabels:       podLabels,
+			ReplicasDesired: 0,
+			Disabled:        false,
+		},
+		watchers: make([]chan struct{}, 0),
+	}
+
+	s.rcs[id] = &entry
+
+	return entry.RC, nil
+}
+
+func (s *fakeStore) List() ([]fields.RC, error) {
+	results := make([]fields.RC, len(s.rcs))
+	i := 0
+	for _, v := range s.rcs {
+		results[i] = v.RC
+		i += 1
+	}
+	return results, nil
+}
+
+func (s *fakeStore) Disable(rc fields.RC) error {
+	entry, ok := s.rcs[rc.Id]
+	if !ok {
+		return util.Errorf("Nonexistent RC")
+	}
+
+	entry.Disabled = true
+	for _, channel := range entry.watchers {
+		channel <- struct{}{}
+	}
+	return nil
+}
+
+func (s *fakeStore) SetDesiredReplicas(rc fields.RC, n int) error {
+	entry, ok := s.rcs[rc.Id]
+	if !ok {
+		return util.Errorf("Nonexistent RC")
+	}
+
+	entry.ReplicasDesired = n
+	for _, channel := range entry.watchers {
+		channel <- struct{}{}
+	}
+	return nil
+}
+
+func (s *fakeStore) Delete(rc fields.RC) error {
+	entry, ok := s.rcs[rc.Id]
+	if !ok {
+		return util.Errorf("Nonexistent RC")
+	}
+
+	if entry.ReplicasDesired != 0 {
+		return util.Errorf("Replicas desired must be 0 to delete.")
+	}
+
+	for _, channel := range entry.watchers {
+		close(channel)
+	}
+
+	delete(s.rcs, rc.Id)
+	return nil
+}
+
+func (s *fakeStore) Watch(rc *fields.RC, quit <-chan struct{}) (<-chan struct{}, <-chan error) {
+	updatesOut := make(chan struct{})
+	entry, ok := s.rcs[rc.Id]
+	if !ok {
+		errors := make(chan error, 1)
+		errors <- util.Errorf("Nonexistent RC")
+		close(updatesOut)
+		close(errors)
+		return updatesOut, errors
+	}
+
+	errors := make(chan error)
+	updatesIn := make(chan struct{})
+	entry.watchers = append(entry.watchers, updatesIn)
+
+	go func() {
+		for _ = range updatesIn {
+			rc.ReplicasDesired = entry.ReplicasDesired
```

Comment:
prone to breakage if we ever add any other fields that can change. make it general?